src/gj.rs

```diff
@@ -30,44 +35,91 @@ enum Instr<'a> {
     },
 }
 
+// FIXME @mwillsey awful name, bad bad bad
+#[derive(Default, Debug, Clone)]
+struct VarInfo2 {
+    occurences: Vec<usize>,
+    intersected_on: usize,
+    size_guess: usize,
+    is_nonparent_input: bool,
+}
+
+struct InputSizes<'a> {
+    cur_stage: usize,
+    // Each stage we're intersecting {set_k}_k
```

Comment:
Having wrote this comment, I'm not sure that it really makes sense out of context. Instead, how about "a map from from stage to vector of costs for each stage, where 'cost' is the largest relation being intersected".